File: backend/src/main/java/project/backend/global/exception/ex/BaseException.java
package project.backend.global.exception.ex;

import lombok.Getter;
import org.springframework.http.HttpStatus;
import project.backend.global.exception.errorcode.ErrorCode;

@Getter
public abstract class BaseException extends RuntimeException {

	private final ErrorCode errorCode;

	public BaseException(ErrorCode errorCode) {
		super(errorCode.message);
		this.errorCode = errorCode;
	}

	public HttpStatus getStatus() {
		return errorCode.status;
	}
}

File: backend/src/main/java/project/backend/global/exception/ex/BaseException.kt
package project.backend.global.exception.ex

import org.springframework.http.HttpStatus
import project.backend.global.exception.errorcode.ErrorCode

abstract class BaseException(val errorCode: ErrorCode) : RuntimeException(
    errorCode.message
) {
    val status: HttpStatus //fixme getStatus() 고려 (추후 없애야하나?)
        get() = errorCode.status
}
